vertical-pod-autoscaler/pkg/apis/poc.autoscaling.k8s.io/v1alpha1/types.go

```diff
@@ -207,3 +207,64 @@ type RecommendedContainerResources struct {
 	// +optional
 	MaxRecommended apiv1.ResourceList `json:"maxRecommended,omitempty" protobuf:"bytes,3,rep,name=maxRecommended,casttype=ResourceList,castkey=ResourceName"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VerticalPodAutoscalerSnapshot is the snapshot of the internal state of VPA that
+// is used for recovery after recommender's restart.
+type VerticalPodAutoscalerSnapshot struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Specification of the snapshot.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status.
+	// +optional
+	Spec VerticalPodAutoscalerSnapshotSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+
+	// Data of the snapshot.
+	// +optional
+	Status VerticalPodAutoscalerSnapshotStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VerticalPodAutoscalerSnapshotList is a list of VerticalPodAutoscalerSnapshot objects.
+type VerticalPodAutoscalerSnapshotList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []VerticalPodAutoscalerSnapshot `json:"items"`
+}
+
+// VerticalPodAutoscalerSnapshotSpec is the specification of the snapshot object.
+type VerticalPodAutoscalerSnapshotSpec struct {
+	// Name of the VPA object that stored VerticalPodAutoscalerSnapshot object.
```

Comment:
There should be content version information.

Reply:
Status/Schema field plays role of content version info. I was thinking whether it should be part of spec - my rationale to put it in status is that this is not configuration, but property of the stored data.

However, I have no strong feelings about that.

Reply:
Version information is stored in status/version